### PR TITLE
added elm hn pwa contrib

### PR DESCRIPTION
**Pull request by skrypte**
hi,
I built a PWA client for HN in Elm. All specs are checked and ok. if, for any reason, you reject my contribution, I will be very glad if you explain what I have to enhance in my code. I wrote this app for educational purpose and I will be happy to receive any feedback or advice.
sincerely

skingrapher

### CODE REVIEW

File: site/_apps/elm-hn-pwa.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+weight: 24 
+title: 'Elm HN PWA'
+github-title: 'skrypte/elm-hn-pwa'
+libraries: "Elm" 
+scaffolding: 'N/A'
+module-bundling: 'N/A'
+service-worker: 'partially workbox.js + application shell + data caching'
```
Comment (addyosmani) [inline site/_apps/elm-hn-pwa.txt]: Could you describe why the entry says 'partially' for workbox support? Generally we keep this fairly high-level (e.g uses workbox + application shell + data-caching)

File: site/_apps/elm-hn-pwa.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+perfomance-patterns: 'preconnect API, inline CSS, minified JS, lazy loading for JSON data'
+server-side-rendering: 'None'
+api: 'hacker-news.firebaseio.com/v0'
+hosting: 'o2switch'
+authors: 
+    - name: 'skingrapher'
+lighthouse: '91/100'
+interactive-em: '4.8s'
+interactive-faster-3g: '3.7s'
+lighthouse-link: https://www.webpagetest.org/lighthouse.php?test=171104_BM_ce030e66965f4ea601c73316b8448f03&run=1
```
Comment (addyosmani) [inline site/_apps/elm-hn-pwa.txt]: Hmm. There's something up with this report as it doesn't appear to have data fetched until after 13 seconds in. I'll take a look at the WebPageTest report to dig into whether this was just a Lighthouse edge-case.

image

File: site/_apps/elm-hn-pwa.txt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+wpt-em-link: https://www.webpagetest.org/result/171104_BM_ce030e66965f4ea601c73316b8448f03/ 
+wpt-faster-3g-link: https://www.webpagetest.org/result/171104_NK_697bd1b5a53d8b3ef8e0afd8a6579a3e/
```
Comment (addyosmani) [inline site/_apps/elm-hn-pwa.txt]: Our general expectations are that data should be fetched and the application is interactive (both the main thread JavaScript and network have settled) within ~5/6 seconds. Looking at WPT, even 20+ seconds in, we're still not getting interactive.

image

Needing to do a separate query for every JSON file appears to be the issue here. There are a lot of them:

image

File: site/_apps/elm-hn-pwa.txt (hunk 1 above)
Comment (skrypte) [reply site/_apps/elm-hn-pwa.txt]: workbox is used only as static cache for assets and application shell

File: site/_apps/elm-hn-pwa.txt (hunk 2 above)
Comment (skrypte) [reply site/_apps/elm-hn-pwa.txt]: well, my app is bloated

File: site/_apps/elm-hn-pwa.txt (hunk 3 above)
Comment (skrypte) [reply site/_apps/elm-hn-pwa.txt]: the issue is here. need to change the code

**Comment by addyosmani** [comment]
Would you like us to continue keeping an eye on this implementation? I noticed that you had some comments in #129.

**Comment by addyosmani** [comment]
As it's been a few quarters since we last heard anything on this PR, we're going to go ahead with closing it. We really appreciate the effort to contribute here. If you would like to try contributing this application back once more, please feel free to open up a new PR addressing previous feedback and we will reconsider it.

**Pull request closed without merging**